pkg/storaged/client.js

```diff
@@ -106,37 +120,40 @@ define([
+        return client.storaged_client.proxies(STORAGED_IFACE_PFX + "." + iface,
+                                              STORAGED_OPATH_PFX,
                                               { watch: false });
     }
 
-    client.storaged_client.watch({ path_namespace: "/org/storaged/Storaged" });
-
-    client.manager = proxy("org.storaged.Storaged.Manager",
-                           "/org/storaged/Storaged/Manager");
-    client.manager_lvm2 = proxy("org.storaged.Storaged.Manager.LVM2",
-                                "/org/storaged/Storaged/Manager");
-    client.manager_iscsi = proxy("org.storaged.Storaged.Manager.ISCSI.Initiator",
-                                 "/org/storaged/Storaged/Manager");
-
-    client.mdraids = proxies("org.storaged.Storaged.MDRaid");
-    client.vgroups = proxies("org.storaged.Storaged.VolumeGroup");
-    client.lvols = proxies("org.storaged.Storaged.LogicalVolume");
-    client.drives = proxies("org.storaged.Storaged.Drive");
-    client.drives_ata = proxies("org.storaged.Storaged.Drive.Ata");
-    client.blocks = proxies("org.storaged.Storaged.Block");
-    client.blocks_ptable = proxies("org.storaged.Storaged.PartitionTable");
-    client.blocks_part = proxies("org.storaged.Storaged.Partition");
-    client.blocks_lvm2 = proxies("org.storaged.Storaged.Block.LVM2");
-    client.blocks_pvol = proxies("org.storaged.Storaged.PhysicalVolume");
-    client.blocks_fsys = proxies("org.storaged.Storaged.Filesystem");
-    client.blocks_crypto = proxies("org.storaged.Storaged.Encrypted");
-    client.iscsi_sessions = proxies("org.storaged.Storaged.ISCSI.Session");
-    client.storaged_jobs = proxies("org.storaged.Storaged.Job");
-
-    client.udisks_client = cockpit.dbus("org.freedesktop.UDisks2");
-    client.udisks_jobs = client.udisks_client.proxies("org.freedesktop.UDisks2.Job",
-                                                      "/org/freedesktop/UDisks2");
+    client.storaged_client.watch({ path_namespace: STORAGED_OPATH_PFX });
+
+    client.manager = proxy("Manager", "Manager");
+    client.manager_lvm2 = proxy("Manager.LVM2", "Manager");
+    client.manager_iscsi = proxy("Manager.ISCSI.Initiator", "Manager");
+
+    client.mdraids = proxies("MDRaid");
+    client.vgroups = proxies("VolumeGroup");
+    client.lvols = proxies("LogicalVolume");
+    client.drives = proxies("Drive");
+    client.drives_ata = proxies("Drive.Ata");
+    client.blocks = proxies("Block");
+    client.blocks_ptable = proxies("PartitionTable");
+    client.blocks_part = proxies("Partition");
+    client.blocks_lvm2 = proxies("Block.LVM2");
+    client.blocks_pvol = proxies("PhysicalVolume");
+    client.blocks_fsys = proxies("Filesystem");
+    client.blocks_crypto = proxies("Encrypted");
+    client.iscsi_sessions = proxies("ISCSI.Session");
+    client.storaged_jobs = proxies("Job");
+
+    if (STORAGED_SERVICE != "org.freedesktop.UDisks2") {
+        client.udisks_client = cockpit.dbus("org.freedesktop.UDisks2");
```

Comment:
It seems strange that the storaged_client can be either: storaged or udisks2, but the udisks_client depends on that. What's the reasoning? Should we just have something like client.client, client.primary_client, client.storage_client or client.dbus_client instead of storaged_client?